tests/test_optimize_qng.py

```diff
@@ -77,12 +78,12 @@ def gradient(params):
             theta_new = opt.step(circuit, theta)
 
             # check metric tensor
-            res = opt.metric_tensor_inv
-            exp = np.diag([4, 4 / (np.cos(theta[0]) ** 2)])
+            res = opt.metric_tensor
+            exp = np.diag([0.25, (np.cos(theta[0]) ** 2)/4])
             assert np.allclose(res, exp, atol=tol, rtol=0)
 
             # check parameter update
-            dtheta = eta * exp @ gradient(theta)
+            dtheta = eta * sp.linalg.pinvh(exp) @ gradient(theta)
```

Comment:
Would it be fine to use ``np.linalg.solve(exp, gradient(theta))`` here as the second term?